--- /Users/slynen/Desktop/gtsam-3.1.0/gtsam/nonlinear/LinearContainerFactor.h	2014-05-08 02:57:09.000000000 +0200
+++ /Users/slynen/Desktop/gtsam-3.1.0/gtsam/nonlinear/LinearContainerFactor_patched.h	2014-10-22 20:23:58.000000000 +0200
@@ -117,7 +117,7 @@
 
   // casting syntactic sugar
 
-  inline bool hasLinearizationPoint() const { return linearizationPoint_; }
+  inline bool hasLinearizationPoint() const { return static_cast<bool>(linearizationPoint_); }
 
   /**
    * Simple checks whether this is a Jacobian or Hessian factor
